cfme/tests/configure/test_start_page.py

```diff
@@ -0,0 +1,87 @@
+    landing_pages = view.visualstartpage.show_at_login.all_options
+    start_page = []
+    for landing_page in landing_pages:
+        start_page.append(landing_page.text)
+    return start_page
+
+
+@pytest.yield_fixture(scope="module")
+def rbac_user(appliance, role):
+    role = role
+    group = Group(description='test{}'.format(faux_value), role=role.name)
+    group.create()
+    group_user = Group('test{}'.format(faux_value))
+    username = 'test{}'.format(faux_value.lower())
+    password = 'test'
+    cred = Credential(principal=username, secret=password)
+    user = User(name='test{}'.format(faux_value),
+                credential=cred, group=group_user)
+    user.create()
+    login_page = navigate_to(appliance.server, 'LoginScreen')
+    login_page.log_in(user)
+    view = navigate_to(Visual, 'All')
+    landing_pages = view.visualstartpage.show_at_login.all_options
+    start_page = []
+    for landing_page in landing_pages:
+        start_page.append(landing_page.text)
+    yield user, landing_pages
+    login_page = navigate_to(appliance.server, 'LoginScreen')
+    login_page.login_admin()
+    user.delete()
+    group.delete()
+
+
+def set_landing_page(value, appliance, user=None):
+    # This list contains the list of pages which show some error or alerts after login.
+    # TODO remove all these pages when BZ is closed.
+    page_list = ['Bottlenecks', 'Automate Log', 'Compute / Containers / Containers']
+    view = navigate_to(Visual, 'All')
+    if (view.visualstartpage.show_at_login.fill(value) and
+            not any(substring in value for substring in page_list)):
+        view.save.click()
+        login_page = navigate_to(appliance.server, 'LoginScreen')
+        if user is None:
+            login_page.login_admin()
+        else:
+            login_page.log_in(user)
+        logged_in_page = navigate_to(appliance.server, 'LoggedIn')
+        if not logged_in_page.is_displayed:
+            return False
+    return True
+
+
+def test_start_page_rbac(appliance, rbac_user):
+    """
+        This test checks the functioning of the landing page; 'Start at Login'
+        option on 'Visual' tab of setting page for custom role based users. This test case doesn't
+        check the exact page but verifies that all the landing page options works properly.
+    """
+    user, start_pages = rbac_user
+    values = [(user, start_page[0]) for start_page in start_pages]
+    for value in values:
+        assert set_landing_page(value[1], appliance, value[0])
+
+
+@pytest.mark.parametrize('start_page', get_landing_page(), scope="module")
+def test_strat_page_admin(start_page, appliance):
+    """
+        This test checks the functioning of the landing page; 'Start at Login'
+        option on 'Visual' tab of setting page for administrator. This test case doesn't
+        check the exact page but verifies that all the landing page options works properly.
+    """
+    assert set_landing_page(start_page, appliance)
```

Comment:
**REQUIRED:** This test feels like it's too magic. What are we testing? There is a little too much abstraction here.

Reply:
When we set the start up page many times we face issues like: 
![startup_page](https://user-images.githubusercontent.com/13481465/32312107-d53b3ed8-bf93-11e7-9620-5698c8a4015c.png)

When this issue arises we are not able to access any page at all, so wanted a test case to keep a track on this issue.

Reply:
Will change the name of the test case

Reply:
We have a bug for this?